cli/cmd/compose/build.go

```diff
@@ -45,7 +45,7 @@ func buildCommand() *cobra.Command {
 }
 
 func runBuild(ctx context.Context, opts buildOptions, services []string) error {
-	c, err := client.New(ctx)
+	c, err := client.NewWithDefaultLocalBackend(ctx)
```

Comment:
I would prefer if we had something like

```go
c, err := client.New(ctx, client.WithContextType(store.LocalContextType)
```

Or something in that vein

Reply:
I didn't want to change the existing API ` client.New(ctx)` all over the place, so I need a function with a different name. We can do `c, err := client.NewWithDefaultBackend(ctx, store.LocalContextType)` in all compose commands. 

(I had this initially, then thought we are not likely to use this with any other backend type, so I published only `client.NewWithDefaultLocalBackend(ctx)`, but passing the `store.LocalContextType` makes it a bit more explicit)

Reply:
Maybe I missed your point, let me quickly check what I can do in the direction of `client.New(ctx, client.WithContextType(store.LocalContextType)`...

Reply:
OK, I missed your point with optional param `...opts ClientOpts`. 
For now, we have only this default backend option, I prefer the explicit call with a known list of typed params. 
If we see we need more options to the client creation, we can change for `client.New(ctx, ... opts ClientOpt)`